src/map.cc

```diff
@@ -435,42 +443,30 @@ Map::init_dimensions() {
   dirs[DirectionUpLeft] = dirs[DirectionLeft] | dirs[DirectionUp];
 
   /* Allocate map */
-  if (tiles != NULL) {
-    delete[] tiles;
-    tiles = NULL;
-  }
-  tiles = new Tile[tile_count]();
-  if (tiles == NULL) abort();
+  deinit();
+
+  game_tiles = new GameTile[tile_count]();
+  if (game_tiles == NULL) abort();
+
+  landscape_tiles = new LandscapeTile[tile_count]();
+  if (landscape_tiles == NULL) abort();
 
   init_spiral_pos_pattern();
 }
 
 /* Copy tile data from map generator into map tile data. */
 void
 Map::init_tiles(const MapGenerator &generator) {
-  for (unsigned int y = 0; y < rows; y++) {
-    for (unsigned int x = 0; x < cols; x++) {
-      MapPos pos_ = pos(x, y);
-      tiles[pos_].height = generator.get_height(pos_) & 0x1f;
-      tiles[pos_].type = (generator.get_type_up(pos_) & 0xf) << 4 |
-        (generator.get_type_down(pos_) & 0xf);
-      tiles[pos_].obj = generator.get_obj(pos_) & 0x7f;
-      if (generator.get_resource_type(pos_) != MineralsNone) {
-        tiles[pos_].resource = (generator.get_resource_type(pos_) & 7) << 5 |
-          (generator.get_resource_amount(pos_) & 0x1f);
-      } else {
-        tiles[pos_].resource = generator.get_resource_amount(pos_);
-      }
-    }
-  }
+  memcpy(landscape_tiles, generator.get_landscape(),
+         rows * cols * sizeof(LandscapeTile));
```

Comment:
We don't use the accessor functions on `MapGenerate` anymore, `get_height()`, `get_type_up()`, etc.? Should these just be removed then?